stl/inc/ranges

```diff
@@ -3768,7 +3772,12 @@ namespace ranges {
 
             _NODISCARD constexpr decltype(auto) operator*() const
                 noexcept(noexcept(_STD get<_Index>(*_Current))) /* strengthened */ {

```

Comment:
I think this needs to check `is_nothrow_move_constructible` if `is_reference_v<range_reference_t<_Base>>` is false. (Same for `operator[]`)

Reply:
Yeah, I was dreading this

Reply:
```cpp
is_nothrow_move_constructible_v<remove_cv_t<tuple_element_t<_Index, range_reference_t<_Base>>>, decltype(_STD get<_Index>(*_Current)))>
```

I am a kind of inclined to remove noexcept for that case

Reply:
> ```c++
> is_nothrow_move_constructible_v<remove_cv_t<tuple_element_t<_Index, range_reference_t<_Base>>>, decltype(_STD get<_Index>(*_Current)))>
> ```

The `remove_cv_t` here is incorrect. If `tuple_element_t</**/>` is `volatile meow`, for instance, you are asking if construction of a `meow` from an rvalue `meow` is non-throwing, instead of asking if construction of a `meow` from an rvalue `volatile meow` is non-throwing (which is what the code actually does). IOW, we need to know `is_constructible_v<remove_cv_t<tuple_element_t</**/>>, tuple_element_t</**/>>` which is equal to `is_constructible_v<tuple_element_t</**/>, tuple_element_t</**/>>` (since cv-quafiliers of the constructed type don't affect constructibility) which we can abbreviate as `is_move_constructible_v<tuple_element_t</**/>>`.

Reply:
I've pushed a commit to fix the conditional `noexcept`s.